### PR TITLE
HTML: document.open() and document's URL

### PR DESCRIPTION
**Pull request by TimothyGu**
For https://github.com/whatwg/html/pull/3946.

Adapts the basic test in https://github.com/web-platform-tests/wpt/pull/10817 for a number of advanced scenarios.

Co-authored-by: Anne van Kesteren <annevk@annevk.nl>

----

Firefox passes all tests; Chrome and Safari only pass the first.

### CODE REVIEW

**Review by domenic — APPROVED** [review]
LGTM with minor suggestions

File: html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url.window.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      childDoc.open();
+      assert_equals(childDoc.URL, blankURL);
+      assert_equals(childWin.location.href, blankURL);
+    });
+    frame.src = "/common/blank.html";
+  });
+  frame.src = frameURL;
+}, "document.open() does not change document's URL (active but not fully active document)");
+
+test(t => {
```
[the remark this answers is not in the document]
Comment (TimothyGu) [reply html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url.window.js]: Hah, well I had that but different browsers do different things when accessing `location.href` on a window that got closed. Firefox returns the empty string I believe, while Chrome returns `undefined` or something similar. Will add a comment.

File: html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url.window.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  const doc = frame.contentDocument;
+
+  // Right now the frame is connected and it has an active document.
+  assert_equals(doc.URL, "about:blank");
+
+  frame.remove();
+
+  // Now the frame is no longer connected. Its document is no longer active.
+  assert_equals(doc.URL, "about:blank");
+  assert_equals(doc.open(), doc);
```
[the remark this answers is not in the document]
Comment (TimothyGu) [reply html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url.window.js]: All the other tests I've added have that assertion for pretty much every `open` call… this is for consistency if nothing else.

File: html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url.window.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (domenic) [reply html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url.window.js]: File a HTML spec issue? :)

File: html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url.window.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (domenic) [reply html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url.window.js]: OK, but in that case it's missing from the first two tests.

**Comment by TimothyGu** [comment]
Ugh, merged the wrong PR. Will have a follow-up PR fixing the review comments.

File: html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url.window.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (TimothyGu) [reply html/webappapis/dynamic-markup-insertion/opening-the-input-stream/url.window.js]: https://github.com/whatwg/html/issues/3959

**Pull request merged**